### PR TITLE
DOP-1471 Product gallery content store panel

### PR DESCRIPTION
**Pull request by ck-makingsense**
Add side panel with integration store on getsettings data

Next steps:

- Add dynamically sorts by integration support
- update product list view 

https://github.com/FromDoppler/doppler-editors-webapp/assets/83654756/3e0b4936-963b-4538-9432-f790de3f27b2

### CODE REVIEW

File: src/implementations/DopplerLegacyClientImpl.ts [hunk 1]
```diff
@@ -214,13 +238,16 @@ function parseDopplerEditorSettings(data: unknown): DopplerEditorSettings {
   // https://github.com/MakingSense/Doppler/pull/10148
   const d = objectOrEmptyObject(data);
   const promotionCodeEnabled = !!d.promotionCodeEnabled;
+
+  // TODO: analyze to use a store parse
```
Comment (leoslopez) [inline src/implementations/DopplerLegacyClientImpl.ts:242]: Should it still be analyzed? Or do we remove this `TODO`?

File: src/components/product-gallery/ProductGalleryUI.test.tsx [hunk 2]
```diff
@@ -377,14 +451,123 @@ describe(ProductGalleryUI.name, () => {
...
+        </AppServicesProvider>
+      </QueryClientProvider>,
+    );
+
+    // Assert
+    const storeItem = screen.getByText("MercadoShops");
+    expect(storeItem.style.color).toEqual("rgb(153, 153, 153)");
+  });
+
+  it("should be a store with no active style", () => {
```
Comment (leoslopez) [inline src/components/product-gallery/ProductGalleryUI.test.tsx:510]: This test has the same description as the previous one.

File: src/components/product-gallery/ProductGalleryUI.test.tsx [hunk 3]
```diff
@@ -377,14 +451,123 @@ describe(ProductGalleryUI.name, () => {
...
+        </AppServicesProvider>
+      </QueryClientProvider>,
+    );
+
+    // Assert
+    const storeItem = screen.getByText("MercadoShops");
+    expect(storeItem.style.color).toEqual("rgb(153, 153, 153)");
+  });
+
+  it("should be a no render a store without productsEnabled false", async () => {
```
Comment (leoslopez) [inline src/components/product-gallery/ProductGalleryUI.test.tsx:530]:
```suggestion
  it("should be a no render a store with productsEnabled false", async () => {
```

File: src/implementations/DopplerLegacyClientImpl.ts (hunk 1 above)
Comment (ck-makingsense) [reply src/implementations/DopplerLegacyClientImpl.ts:242]: deberiamos analizar si agregamos el flag de promocode en cada store, y directamente armamos un parse mas simple, y no dejar en duro el de mercado shop

File: src/components/product-gallery/ProductGalleryUI.tsx [hunk 4]
```diff
@@ -59,24 +59,25 @@ export const ProductGalleryUI = ({
...
-        <ContentList
-          items={items}
-          checkedItemIds={checkedItemIds}
-          toggleCheckedItem={toggleCheckedItem}
-          selectItem={selectItem}
-          hasNextPage={hasNextPage}
-          fetchNextPage={fetchNextPage}
-        />
-      </Content>
+      <div style={{ width: "80%", height: "700px" }}>
```
Comment (leoslopez) [inline src/components/product-gallery/ProductGalleryUI.tsx:62]: I would add a TODO here, something like: `// TODO: replace the hardcoded styles once the rules have been defined on the styles sheet`.

File: src/components/base-gallery/SidePanel.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  value,
+}: {
+  value: string;
+  setValue: (store: string) => void;
+}) => {
+  const [loading, setLoading] = useState(true);
+  const [stores, setStores] = useState<DopplerEditorStore[]>([]);
+  const editorSettings = useGetEditorSettings();
+
+ // TODO resolve missing dependency for useEffect
```
Comment (leoslopez) [inline src/components/base-gallery/SidePanel.tsx]: Is this solved? or will it be solved in another PR?

File: src/components/product-gallery/ProductGalleryUI.test.tsx [hunk 6]
```diff
@@ -7,11 +8,56 @@ import { ModalProvider } from "react-modal-hook";
...
+    },
+  ],
+};
+
+const appServices = {
+  dopplerLegacyClient: {
+    getEditorSettings: () =>
+      Promise.resolve({ success: true, value: editorSettings }),
+  } as unknown,
+  //assetManifestClient,
```
Comment (leoslopez) [inline src/components/product-gallery/ProductGalleryUI.test.tsx:50]: It could be removed, right?

File: src/components/product-gallery/ProductGalleryUI.test.tsx [hunk 7]
```diff
@@ -377,14 +451,103 @@ describe(ProductGalleryUI.name, () => {
...
+        </AppServicesProvider>
+      </QueryClientProvider>,
+    );
+
+    // Assert
+    const storeItem = screen.getByText("MercadoShops");
+    expect(storeItem.style.color).toEqual("rgb(153, 153, 153)");
+  });
+
+  it("should be a no render a store with productsEnabled false", async () => {
```
Comment (leoslopez) [inline src/components/product-gallery/ProductGalleryUI.test.tsx:510]:
```suggestion
  it("Should not render a store with productsEnabled set to false", async () => {
```

File: src/components/product-gallery/ProductGalleryUI.test.tsx [hunk 8]
```diff
@@ -377,14 +451,103 @@ describe(ProductGalleryUI.name, () => {
...
+        </AppServicesProvider>
+      </QueryClientProvider>,
+    );
+
+    // Assert
+    const storeItem = screen.queryByText("VTEX");
+    expect(storeItem).toBeNull();
+  });
+
+  it("should be called setStore when a store item is clicked", async () => {
```
Comment (leoslopez) [inline src/components/product-gallery/ProductGalleryUI.test.tsx:530]:
```suggestion
  it("should call setStore when a store item is clicked", async () => {
```

File: src/components/product-gallery/ProductGalleryUI.test.tsx [hunk 9]
```diff
@@ -377,14 +451,103 @@ describe(ProductGalleryUI.name, () => {
...
+        </AppServicesProvider>
+      </QueryClientProvider>,
+    );
+
+    // Assert
+    const storeItem = screen.getByText("MercadoShops");
+    expect(storeItem.style.color).toEqual("rgb(38, 38, 38)");
+  });
+
+  it("should be a store with no active style", () => {
```
Comment (leoslopez) [inline src/components/product-gallery/ProductGalleryUI.test.tsx:490]: En este caso ningun store estaria seleccionado, no? es decir, todos tendrian el "no active style"?

**Review by leoslopez — APPROVED** [review]
👍

File: src/components/base-gallery/SidePanel.tsx (hunk 5 above)
Comment (ck-makingsense) [reply src/components/base-gallery/SidePanel.tsx]: not yet, in other PR

File: src/components/product-gallery/ProductGalleryUI.test.tsx (hunk 9 above)
Comment (ck-makingsense) [reply src/components/product-gallery/ProductGalleryUI.test.tsx]: that right
